docs/en/observability/monitor-java-app.asciidoc

```diff
@@ -0,0 +1,2005 @@
+[[monitor-java-app]]
+== Monitor a Java application
+
+In this tutorial, you learn how to monitor a Java application using the Elastic
+Observability apps: Logs, Metrics, APM, and Uptime.
+
+[discrete]
+=== What you'll learn
+
+You will learn how to:
+
+- Create a sample Java application.
+- Ingest logs using {filebeat} and view your logs in {kib}.
+- Ingest metrics using the https://www.elastic.co/guide/en/beats/metricbeat/current/metricbeat-module-prometheus.html[Metricbeat
+Prometheus Module] and view your metrics in {kib}.
+- Instrument your application using the https://www.elastic.co/guide/en/apm/agent/java/current/[Elastic APM
+Java agent].
+- Monitor your services using {heartbeat} and view your uptime data in {kib}.
+
+[discrete]
+=== Before you begin
+
+Create a deployment using our hosted {ess} on https://www.elastic.co/cloud/[Elastic Cloud]. The deployment includes
+an {es} cluster for storing and searching your data, {kib} for visualizing and managing
+your data, and an APM server. For more information, see <<set-up-on-cloud,Set up on Cloud>>.
+
+[discrete]
+=== Step 1: Create a Java application
+
+To create the Java application, you require OpenJDK 14 (or higher) and the https://javalin.io/[Javalin]
+web framework. The application will include the main endpoint, an
+artificially long-running endpoint, and an endpoint that needs to poll
+another data source. There will also be a background job
+running. 
+
+. Set up a Gradle project and create the following `build.gradle` file.
++
+[source,gradle]
+----
+plugins {
+  id 'java'
+  id 'application'
+}
+
+repositories {
+  jcenter()
+}
+
+dependencies {
+  implementation 'io.javalin:javalin:3.9.1'
+
+  testImplementation 'org.junit.jupiter:junit-jupiter-api:5.6.2'
+  testRuntimeOnly 'org.junit.jupiter:junit-jupiter-engine:5.6.2'
+}
+
+application {
+  mainClassName = 'de.spinscale.javalin.App'
+}
+
+test {
+  useJUnitPlatform()
+}
+----
++
+. Run the following command.
++
+[source,bash]
+----
+echo "rootProject.name = 'javalin-app'" >> settings.gradle
+
+mkdir -p src/main/java/de/spinscale/javalin
+mkdir -p src/test/java/de/spinscale/javalin
+----
++
+. Install the Gradle wrapper. An easy way to install Gradle is to use
+https://sdkman.io/[sdkman] and run `sdk install gradle 6.5.1`.
++
+. Run `./gradlew clean check`. You should see a
+successful build that has nothing built or compiled yet.
++
+. To create a Javalin server and its first endpoint (the main endpoint), create the
+`src/main/java/de/spinscale/javalin/App.java` file.
++
+[source,java]
+----
+package de.spinscale.javalin;
+
+import io.javalin.Javalin;
+
+public class App {
+    public static void main(String[] args) {
+        Javalin app = Javalin.create().start(7000);
+        app.get("/", ctx -> ctx.result("Appsolutely perfect"));
+    }
+}
+----
++
+. Run `./gradlew assemble`
```

Comment:
This part was confusing for me. I got `bash: ./gradlew: No such file or directory` when I ran this. Do we need to add a step higher up that specifies the user should first run something like `gradle init`?

Reply:
indeed, we need a `gradle wrapper` call higher above that is missing, which would install the wrapper into the project. will add.